static/js/adventure.ts

```diff
@@ -0,0 +1,94 @@
+// @ts-ignore
+// import CKEditorInspector from '@ckeditor/ckeditor5-inspector';
+
+import ClassicEditor from "./ckeditor";
+import { CustomWindow } from './custom-window';
+
+declare let window: CustomWindow;
+
+
+let $editor: ClassicEditor;
+const editorContainer = document.querySelector('#adventure-editor') as HTMLElement;
+const lang = document.querySelector('html')?.getAttribute('lang') || 'en';
+// Initialize the editor with the default language
+if (editorContainer) {
+    initializeEditor(lang);
+}
+
+function initializeEditor(language: string): Promise<void> {
+    return new Promise((resolve, reject) => {
+        if ($editor) {
+            $editor.destroy();
+        }
+
+        ClassicEditor
+            .create(editorContainer, {
+                codeBlock: {
```

Comment:
Maybe in the future we could investigate integrations between CKEditor and CodeMirror. I found this plugin (https://ckeditor.com/cke4/addon/codemirror), but is quite old, but it suggest that perhaps such a thing could be possible with newer versions of CodeMirror and CK

Reply:
Yeah, it's quite old indeed! Let's see how it goes!

Reply:
Do we need an issue for this so we don't forget?